packages/site/site-react/src/preview/BlockPreviewProvider.tsx

```diff
@@ -2,9 +2,26 @@
 
 import { type PropsWithChildren } from "react";
 
+import { useIFrameBridge } from "../iframebridge/useIFrameBridge";
 import { PreviewContext } from "./PreviewContext";
 
+let currentEncryptedContentScope: string | null = null;
+if (typeof window !== "undefined") {
+    // monkey patch the fetch function to include the scope in the headers
+    const originalFetch = window.fetch;
+    window.fetch = async (input, init: RequestInit = {}) => {
+        const headers = new Headers(init.headers);
+        if (currentEncryptedContentScope) {
+            headers.set("X-Block-Preview", currentEncryptedContentScope);
```

Comment:
This sets the header also for requests to 3rd party apis. We could check for relative urls to set it only for them?